packages/block-library/src/button/edit.js

```diff
@@ -277,12 +279,14 @@ function ButtonEdit( props ) {
 				/>
 			</div>
 			<BlockControls group="block">
-				<AlignmentControl
-					value={ textAlign }
-					onChange={ ( nextAlign ) => {
-						setAttributes( { textAlign: nextAlign } );
-					} }
-				/>
+				{ blockEditingMode === 'default' && (
+					<AlignmentControl
```

Comment:
It sounds to me like the `blockEditingMode` should be part of the alignment control component.

Reply:
That's a good option. There are a few controls that have this kind of implementation.

It's a little hard to avoid this issue - https://github.com/WordPress/gutenberg/pull/57941#discussion_r1456713064.

I'll look into if there's an alternative. 👍